src/plopp/backends/matplotlib/image.py

```diff
@@ -1,36 +1,374 @@
+    rasterized:
+        Rasterize the mesh/image if ``True``.
+    **kwargs:
+        Additional arguments are forwarded to Matplotlib's ``pcolormesh``.
     """
-    if (canvas.ax.name != 'polar') and all(
-        (data.coords[dim].ndim < 2)
-        and ((data.coords[dim].dtype == str) or (sc.islinspace(data.coords[dim])))
-        for dim in data.dims
+
+    def __init__(
+        self,
+        canvas: Canvas,
+        colormapper: ColorMapper,
+        data: sc.DataArray,
+        artist_number: int,
+        uid: str | None = None,
+        shading: str = 'auto',
+        rasterized: bool = True,
+        **kwargs,
     ):
-        return FastImage(canvas=canvas, data=data, **kwargs)
-    else:
-        return MeshImage(canvas=canvas, data=data, **kwargs)
+        check_ndim(data, ndim=2, origin='MeshImage')
+        self.uid = uid if uid is not None else uuid.uuid4().hex
+        self._canvas = canvas
+        self._colormapper = colormapper
+        self._ax = self._canvas.ax
+        self._data = data
+        self._shading = shading
+        self._rasterized = rasterized
+        self._kwargs = kwargs
+        self._optimized_mode = _suitable_for_fast_image(self._canvas, self._data)
+
+        self._string_labels = {}
+        self._bin_edge_coords = {}
+        self._raw_coords = {}
+        to_dim_search = {
+            k: {'dim': self._data.dims[i], 'var': self._data.coords[self._data.dims[i]]}
+            for i, k in enumerate('yx')
+        }
+
+        self._update_coords()
+        self._dim_1d, self._dim_2d = _get_dims_of_1d_and_2d_coords(to_dim_search)
+
+        self._image = self._make_image()
+
+        self._colormapper.add_artist(self.uid, self)
+        self._update_colors()
+
+        for xy, var in self._string_labels.items():
+            getattr(self._ax, f'set_{xy}ticks')(np.arange(float(var.shape[0])))
+            getattr(self._ax, f'set_{xy}ticklabels')(var.values)
+
+        self._canvas.register_format_coord(self.format_coord)
+
+    def _update_coords(self) -> None:
+        self._data_with_bin_edges = sc.DataArray(data=self._data.data)
+        for i, k in enumerate('yx'):
+            self._raw_coords[k] = self._data.coords[self._data.dims[i]]
+            self._bin_edge_coords[k] = coord_as_bin_edges(
+                self._data, self._data.dims[i]
+            )
+            self._data_with_bin_edges.coords[self._data.dims[i]] = (
+                self._bin_edge_coords[k]
+            )
+            if self._data.coords[self._data.dims[i]].dtype == str:
+                self._string_labels[k] = self._data.coords[self._data.dims[i]]
+
+        if self._optimized_mode:
+            self._xmin, self._xmax = self._bin_edge_coords["x"].values[[0, -1]]
+            self._ymin, self._ymax = self._bin_edge_coords["y"].values[[0, -1]]
+            self._dx = np.diff(self._bin_edge_coords["x"].values[:2])
+            self._dy = np.diff(self._bin_edge_coords["y"].values[:2])
```

Comment:
Ideally, you wouldn't define attributes of `self` outside of `__init__`. And I don't like that these attrs only exist for the optimized mode.
Can you use composition to have a `FastImage` or `MeshImage` object inside `Image`? This might also get rid of the `if self._optimized_mode:` checks scattered everywhere.